fuel-tx/src/tests/valid_cases/transaction.rs

```diff
@@ -136,6 +116,142 @@ fn maturity() {
     assert_eq!(CheckError::TransactionMaturity, err);
 }
 
+#[test]
+fn witness_limit_script_no_limit_works() {
+    let rng = &mut StdRng::seed_from_u64(8586);
+
+    let block_height = 1000.into();
+
+    TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
+        .add_random_fee_input()
+        .finalize()
+        .check(block_height, &test_params())
+        .expect("Failed to validate script");
+}
+
+#[test]
+fn witness_limit_create_no_limit_works() {
+    let rng = &mut StdRng::seed_from_u64(8586);
+
+    let block_height = 1000.into();
+    let bytecode = vec![];
+    TransactionBuilder::create(bytecode.clone().into(), rng.gen(), vec![])
+        .add_random_fee_input()
+        .finalize()
+        .check(block_height, &test_params())
+        .expect("Failed to validate tx create");
+}
+
+#[test]
+fn witness_limit_script_limit_works() {
```

Comment:
This should probably be 2 tests. Then you'd have something like:

```
// given
let witness_size = Signature::LEN + vec![0u8; 0].size_static();

// when
let limit = witness_size;
let res = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
        .add_random_fee_input()
        .witness_limit(limit as u64)
        .finalize()
        .check(block_height, &test_params());
        
// then
assert!(res.is_ok());
```
And the other can just do the same but with:
```
let limit = witness_size - 1;
```
with the same `assert_eq!`

The same goes for all the following tests.